modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java

```diff
@@ -0,0 +1,251 @@
+package org.wso2.identity.integration.test.oidc;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.identity.integration.test.oidc.bean.OIDCApplication;
+import org.wso2.identity.integration.test.oidc.bean.OIDCUser;
+import org.wso2.identity.integration.test.util.Utils;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OIDCScopeValidationTest extends OIDCAbstractIntegrationTest {
+
+    protected OIDCUser user;
+    protected String accessToken;
+    protected String sessionDataKeyConsent;
+    protected String sessionDataKey;
+    protected String authorizationCode;
+
+    CookieStore cookieStore = new BasicCookieStore();
+
+    protected HttpClient client;
+    protected List<NameValuePair> consentParameters = new ArrayList<>();
+    OIDCApplication playgroundApp;
+    ServiceProvider serviceProvider;
+    private String claimsToGetConsent;
+    private String scopes;
+
+    private static final String INTERNAL_LOGIN_SCOPE = "internal_login";
+    private static final String SYSTEM_SCOPE = "SYSTEM";
+    private static final String INTERNAL_USER_MGT_UPDATE_SCOPE = "internal_user_mgt_update"; // admin permission.
+    private static final String INVALID_SCOPE = "internal_user_update"; // This scope does not exist.
+
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit(Method method) throws Exception {
+
+        super.init();
+        String scope = OIDCUtilTest.role;
+        if (method.getName().equalsIgnoreCase("testScopeValidationWithAdminUser")) {
+            scope = "admin";
+        }
+        initUser(scope);
+        createUser(user);
+        userInfo.setUserName(user.getUsername());
+        userInfo.setPassword(user.getPassword());
+
+        playgroundApp = initApplication();
+        serviceProvider = createApplication(new ServiceProvider(), playgroundApp);
+        client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+
```

Comment:
redundant blank lines.
Let's check other places as well

Reply:
Fixed with https://github.com/wso2/product-is/pull/13004/commits/6215fdfab95c193b90112d9a983ea3bd0113ee46